### DISCUSSION TITLE
Customising service form in SQUEST

### DISCUSSION THREAD

**Discussion by pravyn** [HewlettPackard/squest#683] [Q&A]
Hello,

i want to create service form with the fields as below

1) VM name (text box)
2)location (drop down)
3) vm architecture (checkbox)
---- etc.

We already have play books to create virtual machines so trying to understand how to create an operation (for creating VM) with the above fields and pass them as variables to the playbook.

Sorry, to post this question. I' am sure you might have documented but unable to find it out.

Thanks,
Praveen

**Comment by Sispheor** [comment]
I'm not sure to understand. You already have a playbook to create a VM?
This playbook is declared as job template in Tower/AWX?

**Reply by EliasBoulharts** [reply]
The survey is declared in your Job Template on the AWX side, after which you can create an operation and define certain fields as administration fields on the Squest side.

**Reply by pravyn** [reply]
@EliasBoulharts  @Sispheor 

Sorry for the confusion.
I have survey created for VM creation with "VMname" and "location" as fields. But how to have similar fields on the form ? currently i can see only "Squest Instance name" & "comment".


<img width="1175" alt="image" src="https://github.com/HewlettPackard/squest/assets/19923046/33354a97-ef67-4240-8923-700e381250fa">


<img width="693" alt="image" src="https://github.com/HewlettPackard/squest/assets/19923046/1bbd87ef-e815-45c9-bae8-7a5b9056cc83">


<img width="984" alt="image" src="https://github.com/HewlettPackard/squest/assets/19923046/4bbc47dc-9ba1-4919-b2fc-af4bc4dce84f">

**Reply by Sispheor** [reply]
Your survey is disabled
![image](https://github.com/HewlettPackard/squest/assets/2573405/c3f905f1-86fd-498c-b521-ea3b9031f9b5)

**Reply by pravyn** [reply]
I have enabled the survey now in AWX. Where i need to add the survey fields in order to show in "Squest Fields".

**Reply by Sispheor** [reply]
Use the sync button on squest job template

**Reply by pravyn** [reply]
Thanks again, able to see the fields now from AWX survey after sync. Also, can we remove "Squest Fields" from the form, just want to have "Service Fields" while submitting the request.


<img width="1021" alt="image" src="https://github.com/HewlettPackard/squest/assets/19923046/0905dca2-7d15-449a-bc9a-05d89c96ecc5">

**Reply by Sispheor** [reply]
no squest fields are mandatory. Especially the instance name as it's what will be created for the end user to be able to apply day 2 operations later. Please read the concept doc.
